src/components/Insertion/SearchBar/index.tsx

```diff
@@ -0,0 +1,31 @@
+import { useState } from 'react';
+import { useNavigate } from 'react-router';
+
+const SearchBar: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const navigate = useNavigate();
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      e.preventDefault();
+      navigate(`/results?query=${query}`);
+    }
+  };
+
+  return (
+    <div>
```

Comment:
giusto per fartelo vedere, qui possiamo usare un `<form>` anziche un `<div>`. il discorso dei form e' molto complicato e lungo, e come hai fatto tu funziona chiaramente, ma ci sono tot cose che vanno tenute a mente.

nel tuo caso funziona perche react con useState ti permette di sicronizzare l'input a uno stato, a cui hai accesso quando gestisci l'evento di tasto premuto con Enter. ma come funzionerebbe se non avessi react? se avessi solo html senza javascript?

l'html come linguaggio si continua ad evolvere per permettere di creare interfacce e comportamenti sempre migliori, e i form in particolare (e la ricerca) esistono dagli anni '90


con un HTML cosi:
```tsx
<form action="/results">
  <input type="search" name="query" />
</form>
```

hai esattamente lo stesso comportamento che hai fatto tu "a mano", con i benefici che:
- e' accessibile. gli screen reader leggono un form e si comportano diversamente; gli utenti ricevono un autocomplete nei campi se hanno gia compilato e infine non hai bisogno di gestire a mano Enter, e' un comportamento nativo del browser che premere Enter sull'ultimo input di un form ti submitta il form
- usa un decimo del codice
- (non importante adesso) funziona senza javascript*

c'e un piccolissimo MA su questo approccio, che dovrebbe essere risolto usando `<Form>` di react router anziche il <form> nativo html, il resto e' uguale.

se vuoi approfondire queste cose tipo cosa cambia tra Form di react router e quello nativo possiamo sentirci